units/host_monitoring_external_termination.go

```diff
@@ -136,10 +137,19 @@ func handleExternallyTerminatedHost(ctx context.Context, id string, env evergree
 		if cloudStatus != cloud.StatusTerminated && (h.UserHost || h.StartedBy != evergreen.User) {
 			return false, errors.New("non-agent host is not already terminated and should not be terminated")
 		}
-
+		if h.RunningTask != "" {
+			if err := task.AddHostCreateDetails(h.RunningTask, h.Id, errors.New("host was externally terminated")); err != nil {
```

Comment:
Just realized: h.RunningTask is set for the host that's running the task; we don't want the host that is _running_ the task but rather the host that was _created by_ the task (same below). Looking at what we query for when looking for uphosts, I think we instead want to use `h.SpawnOptions.TaskId` ( and we can check h.SpawnOptions.SpawnedByTask)